server/routerlicious/packages/services-core/src/storageNameProvider.ts

```diff
@@ -0,0 +1,9 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IStorageNameProvider {
+    getStorageName(tenantId: string, documentId: string): Promise<string>;
+    createStorageName(tenantId: string, documentId: string): Promise<string>;
```

Comment:
What's the intended difference between get/create storage name? The current implementation is the same for both (get tenant storageName from Tenant DB), but I'm assuming there is a plan for what create would do differently from get? Perhaps add some JSDoc comments describing these?

It's also possible that I'm just confused by the phrase "create storage name", which seems to imply that a resource is being created. My understanding is that we'll be getting a storage name from a list of some sort, but not actually creating a new one for the given document.

Reply:
Perhaps `createStorageName()` means calling some sort of load balancer to find out what specific storage should be used for the current documentId, based on the tenant (and its region)? In that case, I am curious as to why we would need the documentId here

Reply:
@znewton The `getStorageName` is **retrieve** a storage name for a existing document while `createStorageName` is **Assign** a storage name for a new document. Maybe it is better for me just change to retrieve... and assign...
`StorageNameProvider` is a default implementation of `IStorageNameProvider` for OSS, where FRS will have its own implementation and will inject its implementation into the resourcefactory via `customization?` object. We have the similar done for document collection migration.

@hedasilv While assigning, there are couple of different strategy for load balancing, I am thing of:
1. round robin
2. shortest ping latency
3. documentId hashing 
4. ...

So maybe I am over designing, that is why I was thinking to keep `documentId` , which will be used for strategy 3

Reply:
Changed to `retrieve`/`assign` name pair for storage name

Reply:
I had a similar question around `createStorageName`. @zhangxin511, could you share the thinking behind this design?